cmd/ketch/platform_add_test.go

```diff
@@ -0,0 +1,113 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"io/ioutil"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/shipa-corp/ketch/internal/api/v1beta1"
+)
+
+func TestPlatformAdd(t *testing.T) {
+	sf := func(s string) string { return "-" + s }
+	tt := []struct {
+		name    string
+		args    []string
+		testFn  func(obj runtime.Object) error
+		called  bool
+		wantErr bool
+	}{
+		{
+			name: "happy path",
+			args: []string{
+				"java",
+				sf(imageShortFlag),
+				"somerepo/someimage:latest",
+			},
+			called: true,
+			testFn: func(obj runtime.Object) error {
+				p, ok := obj.(*v1beta1.Platform)
+				require.True(t, ok)
+				require.Equal(t, "java", p.ObjectMeta.Name)
+				require.Equal(t, "somerepo/someimage:latest", p.Spec.Image)
+				return nil
+			},
+		},
+		{
+			name: "official image",
+			args: []string{
+				"java",
+			},
+			called: true,
+			testFn: func(obj runtime.Object) error {
+				p, ok := obj.(*v1beta1.Platform)
+				require.True(t, ok)
+				require.Equal(t, "java", p.ObjectMeta.Name)
+				require.Equal(t, "shipasoftware/java:latest", p.Spec.Image)
+				return nil
+			},
+		},
+		{
+			name: "ambiguous commands",
+			args: []string{
+				"java",
+				sf(imageShortFlag),
+				"somerepo/someimage:latest",
+				sf(dockerfileShortFlag),
+				"/home/bob/Dockerfile",
+			},
+			wantErr: true,
+		},
+		{
+			name: "unknown official platform",
+			args: []string{
+				"pascal",
+			},
+			wantErr: true,
+		},
+		{
+			name: "client failed",
+			args: []string{
+				"java",
+				sf(imageShortFlag),
+				"somerepo/someimage:latest",
+			},
+
+			testFn: func(obj runtime.Object) error {
+				return errors.New("xxx")
+			},
+			wantErr: true,
+		},
+	}
+
+	for _, tc := range tt {
+		t.Run(tc.name, func(t *testing.T) {
+			cli := &paCreateMock{tc.testFn, false}
+			cmd := newPlatformAddCmd(cli, ioutil.Discard)
+			cmd.SetArgs(tc.args)
+			cmd.SetOut(ioutil.Discard)
+			err := cmd.Execute()
+			if tc.wantErr {
+				require.NotNil(t, err)
+				return
+			}
+			require.Nil(t, err)
+			require.Equal(t, tc.called, cli.called)
+		})
+	}
+}
+
+type paCreateMock struct {
+	f      func(o runtime.Object) error
```

Comment:
because it's a mock for `resourceCreator`, do you think we can move it to a separate file and rename to `resourceCreatorMock` then we can use it with other commands?